Standardize Docs: app, auto-updater, styleguide

jlord opened the pull request:
I am embarking an effort to help out with documentation maintenance :books: This PR is a WIP and the first step: a once-over to standardize the format of the docs.

**Standardization tasks:**
- Max. one `h1` per page.
- Use `bash` instead of `cmd` (because of syntax highlighter).
- Change doc titles to their object name format (i.e. `browser-window` → `BrowserWindow`).
- No headers following headers, add at least one sentence description.
- Methods wrapped in `code` ticks.
- Events wrapped in singe 'quotation' marks.
- No nesting more than 2 deep (unfortunately because of markdown renderer).
- Add section titles Events, Class Methods and Instance Methods.
- Use 'will' over 'would' when describing outcomes.
- Events and methods are `h3`.
- Optional arguments written as `function (required[, optional])`.
- Optional arguments are denoted when called out in list.
- Line length is 80-column wrapped.
- Platform specific methods are noted in italics following method header.

And a few other smaller things like consistent periods in lists, phrasing, specifying 'Returns' for returned values...

**Update Docs Website**

Additionally I'll be improving some of the styles on the electron.atom.io/docs website. For the sake of visualizing some of these text changes, here's a WIP screenshot of how they'd look on the site:

<img width="881" alt="screen shot 2015-08-20 at 1 10 27 pm" src="https://cloud.githubusercontent.com/assets/1305617/9381995/e0159f78-473c-11e5-861a-7f5694afb3ef.png">

---

<img width="871" alt="screen shot 2015-08-20 at 1 10 54 pm" src="https://cloud.githubusercontent.com/assets/1305617/9382017/fa0462a2-473c-11e5-8772-06bd0619c0bd.png">

---

<img width="873" alt="screen shot 2015-08-20 at 1 11 10 pm" src="https://cloud.githubusercontent.com/assets/1305617/9382018/ff012c5e-473c-11e5-9b62-1e9b281602d6.png">

---

Future things (in other PRs): Directories for translations, more code examples, diagrams...

kevinsawicki commented:
@OlsonDev thanks so much for the feedback on this. In the future it might be more helpful to comment on lines from the `Files Changed` tab in the pull request versus the individual commits.

That way it is easier to associate the comments with the lines they are referring to. Below is a screenshot of how it currently looks when commenting on individual commits. It can be a little confusing as to what they refer to since they are attached to the commits instead of individual lines in the diff.

<img width="811" alt="screen shot 2015-08-20 at 9 36 05 am" src="https://cloud.githubusercontent.com/assets/671378/9389282/09b55d6e-471f-11e5-8c13-1553326bcea2.png">

OlsonDev commented:
Ah, my bad; I didn't know that was there. Sorry about that. I'll keep that in mind next time.

zcbenz commented:
I :heart: this!

Here are some more places that may need to be standardized:
* Some API's options have optional fields, we should probably add `(optional)` in the descriptions.
* Some APIs are platform-dependent, maybe we can find a better way to say that instead of putting `Note: This API is OS X only.` in the descriptions.
* Some documents are 80-column wrapped, but some are not, we should make them have same format.

kevinsawicki commented:
> Some APIs are platform-dependent, maybe we can find a better way to say that instead of putting Note: This API is OS X only. in the descriptions.

:+1: This would also be great to clearly label in the generated HTML

kevinsawicki commented:
> Some files are 80-column wrapped, but some are not, we should make them have same format.

I like striving for 80 column wrapped unless people really think it is too much overhead.

jlord commented:
Yes! Totally agree with those points @zcbenz. In one place I did do a **required** in bold and it was something I meant to mention as well, definitely want to standardize that. 

Awesome! Gonna upate :+1:

jlord commented:
> Some APIs are platform-dependent

@zcbenz we could write the markdown such like:

```markdown
### `app.setUserTasks(tasks)` _Windows_
```

And use css to style it so on the site it's like this:

<img width="577" alt="screen shot 2015-08-22 at 2 59 23 pm" src="https://cloud.githubusercontent.com/assets/1305617/9424049/69cafb3a-48de-11e5-8594-27424baf2ced.png">

Of course on the raw Markdown renders like below, which perhaps we can live with?

<img width="596" alt="screen shot 2015-08-22 at 3 00 11 pm" src="https://cloud.githubusercontent.com/assets/1305617/9424055/8f59ce58-48de-11e5-94ce-5ff2dec1d615.png">

zcbenz commented:
:+1:

jlord commented:
Ok I've added the things @zcbenz mentioned to the outline in the PR body and the styleguide file that this PR adds. I've also changed the title of the PR because I'm going to do this docs work in batches with just a few files each so that the PRs are more digestible. 

This one should be ready for final review then.

zcbenz commented on docs/api/app.md:
```diff
@@ -291,25 +315,25 @@ Sets the string to be displayed in the dock’s badging area.
 
 **Note:** This API is only available on OS X.
```
Should we remove all the `This API is only available on xxx.` paragraphs?

jlord replied:
Yeah that sounds good. I'll add a note to the top that explains platform specific APIs :+1:

zcbenz commented on docs/styleguide.md:
````diff
@@ -0,0 +1,78 @@
+- Optional arguments are denoted when called out in list.
+- Line length is 80-column wrapped.
+- Platform specific methods are noted in italics following method header.
+ - ```### `method(foo, bar)` _Mac_```
````
We should use "OS X" instead of "Mac" here, there are probably other places need to be changed too.

jlord replied:
Whoops, will update :+1:

kevinsawicki commented:
Looks good to me, my vote is to :ship:

jlord commented:
Wooooo!

thomasjo commented:
Awesome :sparkling_heart:
